### ISSUE TITLE
Unexpected bytes in ByteBuffer of a TCP connection

### ISSUE THREAD

**Issue by Torgrimm** [helios-io/helios#34]
Hello,

I'm trying to utilize Helios in my application to get rid of performance limitations of the former native .Net TcpClient/UdpClient based implementation when connecting with a lot of remote devices. Setting up the TCP connection worked after some try and error.

I've wrote a custom MessageDecoder implementation to reflect our protocol. however, at some point i always get malformed/unexpected bytes in the stream.
Sometimes the data looks like the devices UDP discovery broadcast. Sometimes it could be DHCP discover broadcast messages. But how could they end up in the TCP stream?

I've seen in the LengthFieldFrameBasedDecoder there is a mechanism for discarding too long frames. Is tat intended to fix the behavior, I've discovered?

**Comment by Aaronontheweb** [comment]
@Torgrimm yes, it's possible due to the unpredictable nature of byte buffering from the socket to have partial messages and so your decoder has to be equipped for doing that. The logic built into the `LengthFrameDecoder` has some handling for detecting malformed messages but it's pretty weak to be honest.

I'm not going to be making any additional fixes to the v1.4 branch of Helios - working on publishing the 2.0 spec which should make it a lot easier to handle these sorts of issues in the future.

**Comment by Torgrimm** [comment]
Hello Aaron
thank you for your confirmation. When do you expect to release Helios 2.0?
Also it seems, it is not guaranteed that Streamed Connection packages (TCP) are decoded in sequence number ordering. Can you confirm this as well?
